internal/apex/wg.go

```diff
@@ -107,6 +107,10 @@ func (ax *Apex) handlePeerDelete(peerListing []models.Peer) error {
 		if inPeerListing(peerListing, p) {
 			continue
 		}
+		// skip if the key is the same as the node itself
+		if ax.keyCache[p.DeviceID] == ax.wireguardPubKey {
+			continue
+		}
```

Comment:
We would only hit this if our own peer disappeared from the peer listing, right?  This should never happen. I would rather hold off and fix whatever the root issue is that you're hitting.

If you do want to merge this, I'd at least log an `ERROR` about it since it should never happen.